integration-tests/artifacts/__tests__/static-queries.js

```diff
@@ -0,0 +1,173 @@
+  })
+
+  test(`are written correctly in jsx`, async () => {
+    const queries = [titleQuery, ...globalQueries]
+    const pagePath = `/jsx/`
+
+    const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+    expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  })
+
+  test(`are written correctly in tsx`, async () => {
+    const queries = [titleQuery, ...globalQueries]
+    const pagePath = `/tsx/`
+
+    const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+    expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  })
+
+  test(`are written correctly in typescript`, async () => {
+    const queries = [titleQuery, ...globalQueries]
+    const pagePath = `/typescript/`
+
+    const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+    expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  })
+
+  test(`are written correctly when nesting imports`, async () => {
+    const queries = [titleQuery, authorQuery, ...globalQueries]
+    const pagePath = `/import-import/`
+
+    const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+    expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  })
+
+  test(`are written correctly when nesting dynamic imports`, async () => {
+    const queries = [titleQuery, ...globalQueries]
+    const pagePath = `/dynamic-dynamic/`
+
+    const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+    expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  })
+
+  test(`are written correctly when nesting a dynamic import in a regular import`, async () => {
+    const queries = [titleQuery, authorQuery, ...globalQueries]
+    const pagePath = `/import-dynamic/`
+
+    const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+    expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  })
+
+  test(`are written correctly when nesting a regular import in a dynamic import`, async () => {
+    const queries = [titleQuery, ...globalQueries]
+    const pagePath = `/dynamic-import/`
+
+    const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+    expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  })
+
+  // test(`are written correctly when using wrapRootElement`, async () => {
+  //   const queries = [titleQuery]
+  //   const pagePath = `/dynamic-import/`
+
+  //   const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+  //   expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  // })
+
+  // test(`are written correctly when using wrapPageElement`, async () => {
+  //   const queries = [titleQuery]
+  //   const pagePath = `/dynamic-import/`
+
+  //   const { staticQueryHashes } = await readPageData(publicDir, pagePath)
+
+  //   expect(staticQueryHashes.sort()).toEqual(queries.map(hashQuery).sort())
+  // })
```

Comment:
can we add 2 different queries, and make them global and uncomment those 2 tests?

Reply:
The thing is that those will be included in all of the pages (and hence tests) above. Which is why I commented these out.